api/auth.proto

```diff
@@ -44,3 +46,73 @@ message AuthAction {
   // List of rules
   repeated Rule rule = 2;
 }
+
+service Authorization {
+  rpc Check(CheckRequest) returns (CheckResponse);
+}
+
+message CheckRequest {
+  // Address of the downstream source of the request.
+  Address source = 1;
+
+  // Address of the upstream destination of the request.
```

Comment:
@mandarjog I want to avoid munging network layer attributes with those from higher layers, like the host header.  This is more like `destination.ip` and `destination.port` in Mixer attribute language.